### DISCUSSION TITLE
How to add Serializable exceptions?

### DISCUSSION THREAD

**Discussion by Errechydy** [serverpod/serverpod#1964] [Q&A] [answered]
In this [documentation page](https://docs.serverpod.dev/concepts/exceptions) that talks about Error handling and exceptions, it says that you can add Serializable exceptions the same way you add models but change `class` to `exception`, everything is clean my question is How to include those exception so their classes can be generated and used?  because you don't want to mix them with normal models.
I put them in a folder at the same location as `models` folder like this:
`_server/lib/exceptions/endpoint-name/exception1.yaml`
Inside `exceptions` folder i made sub folders so i can put all exceptions related to a specific endpoint in one place.

In this case how can i include those  all those exceptions (yaml) so their classes can be generated and used?

**Comment by Isakdl** [comment, marked as answer]
The exceptions also need to be placed in the `models` folder to be picked up by the generate command.

**Reply by Errechydy** [reply]
Thanks, that's what i ended up doing, but is there a way to put them in a specific folder, and not mixed up with the normal `models`?
If that's not possible at the moment then that would be a great feature to add in the upcoming versions, it will help a lot to organize files.

**Reply by Isakdl** [reply]
Currently not possible but would be a great addition!
